src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,37 @@
      * Андрей - 600
      * София - 900</p>
      */
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate startDate = LocalDate.parse(dateFrom, DATE_FORMAT);
+        LocalDate endDate = LocalDate.parse(dateTo, DATE_FORMAT);
+
+        if (startDate.isAfter(endDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        String[] dataOnParts = new String[4];
```

Comment:
4 is magic number, let's make it constant and let's create this variable right before we will use it, it will improve readability